### DISCUSSION TITLE
Welcome to jsoncpp Discussions!

### DISCUSSION THREAD

**Discussion by cdunn2001** [open-source-parsers/jsoncpp#1255] [General]
Let's see if the new GitHub "Discussions" feature is useful to us.

## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pramodpbk** [comment]
Hi, I am using jsonCpp and it is seems causing segmentation fault in one of platform.
callstack is as below,it works fine with various other linux platform but with this particular case it causing seg fault when Reader is created. Please guide us if you have any insight into it.
#0 0xb6de5e98 in dlfree () from libc.so
#1 0x00371dc8 in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string() ()
#2 0x0004ae48 in acp::Json::Reader::~Reader() () at <path>/include/jsoncpp/json.h:1240

**Reply by BillyDonahue** [reply]
It looks like a reference lifetime issue on your end. It's in the Reader's destructor not constructor. Would need more context to help.

**Reply by pramodpbk** [reply]
Thanks for response.
we are doing something like below.
**********************
 1.std::cout << << "test1;
2.Json::Reader reader;
3.  std::cout << << "test2;
4. bool parsingSuccessful = reader.parse(file, root_, false);
5.  std::cout << << "test3;
****************************
It is cuasing seg fault at 2. where reader instance is created.

When we replace parser code as below,with redirection "file >> root_";, it continues and crashes at next instance of reader / fastwriter.

 1.std::cout << << "test1;
2. //json::Reader reader;
3.  std::cout << << "test2;
4. //bool parsingSuccessful = reader.parse(file, root_, false);
5. file >> root_;
5.  std::cout << << "test3;

**Reply by BillyDonahue** [reply]
Something else is wrong with your system by the time line 2 runs.
You should debug it on your own, figure out what stack is active when the segfault happens, and update this with a more precise description of steps in C++ (the pseudocode isn't doing it for me).

I feel like there must be some formatting problem in your example.  The `std::cout << << "test1;` etc won't build.
And you're saying you get a segfault at line 2, where you are just making a `Json::Reader`?

**Comment by Lynn0601** [comment]
Hi,I'm only the g++ 5.4 version.Does there is a old version supported it?Thanks.

**Reply by cdunn2001** [reply]
There is an older version on the `0.y.z` branch. Totally unsupported, but it should still work.
